google_compute_engine/network_utils.py

```diff
@@ -38,11 +44,16 @@ def _CreateInterfaceMap(self):
       dict, string MAC addresses mapped to the string network interface name.
     """
     interfaces = {}
-    for interface in os.listdir('/sys/class/net'):
+    for interface in netifaces.interfaces():
       try:
-        mac_address = open(
-            '/sys/class/net/%s/address' % interface).read().strip()
-      except (IOError, OSError) as e:
+        mac_address = (
+            netifaces.ifaddresses(interface)[netifaces.AF_LINK][0]['addr'])
```

Comment:
I see. I changed the approach then, it shouldn't be any exceptions raised by netifaces in normal cases (only in extreme cases as a lack of memory for example), so I used the code you suggested and removed the try/catch statement.